### PR TITLE
No UI about dialog

### LINKED ISSUE #7842: Desklow window crashes/quits when opening `Help -> About Deskflow...` menu item
### Project

Deskflow

### Deskflow version number

1.17.0.166

### Build type

Deskflow package (downloaded from Deskflow)

### Operating systems (OS)

- [X] Windows
- [ ] macOS
- [ ] Linux (X11)
- [ ] Linux (Wayland)
- [ ] BSD-derived
- [ ] Other (please specify)

### Wayland on Linux

- [ ] I have reviewed the Wayland [known issues](https://github.com/deskflow/deskflow/discussions/7499) and my issue is new
- [ ] I am not using Wayland on Linux

### Signing on macOS

- [ ] I have authorized the app to run on my Mac
- [ ] I am not using macOS

### Continuous build

- [X] I have tried the latest continuous build and the issue persists
- [ ] I am unable to try the latest continuous build

### OS versions/distros

Windows 10

### Deskflow configuration

Windows 10 server
macOS 15.1 client

### What steps will reproduce the problem?

1. Open Deskflow main window
2. Go into Help menu -> About Deskflow...
3. Crash

### Log output

```shell
Faulting application name: deskflow.exe, version: 1.17.0.166, time stamp: 0x67237c3e
Faulting module name: MSVCP140.dll, version: 14.32.31326.0, time stamp: 0x7255e0ca
Exception code: 0xc0000005
Fault offset: 0x0000000000013278
Faulting process id: 0x2ae0
Faulting application start time: 0x01db2bfccf63de9d
Faulting application path: C:\Program Files\Deskflow\deskflow.exe
Faulting module path: C:\Windows\SYSTEM32\MSVCP140.dll
Report Id: 80d9a7ab-22b8-4b59-91f8-704aebfe9894
Faulting package full name: 
Faulting package-relative application ID:
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by sithlord48**
fixes #7842 
- Fix light/ dark detection

New About Dialog with no UI file moved to `/src/gui/src/dialogs/`

Dark with my theme icon 
Screenshot_20241031_133634
Light with theme icon
Screenshot_20241031_133703

Dark with fallback icon
Screenshot_20241031_142225

Light with fallback icon

Screenshot_20241031_142302

### CODE REVIEW

**Review by nbolton — COMMENTED** [review]
Looks like the dark logo is being displayed in your screenshot. I'll test on mine and see if I get the same bug.

**Comment by sithlord48** [comment]
> Looks like the light logo (i.e. logo that should be used in light mode) is being displayed in your screenshot. I'll test on mine and see if I get the same bug.

its possible i put the backwards lmk

**Comment by nbolton** [comment]
Ok, now it's working but maybe logo is a tad fuzzy? 

Hmm, also the copy button isn't showing for me. I'm on GNOME, not sure if relevant.

image

**Comment by nbolton** [comment]
Some improvements I'd like to make, too many changes to make it a GitHub PR suggestion comment.
```diff
diff --git a/cspell.json b/cspell.json
index 166551ce..3e6be29f 100644
--- a/cspell.json
+++ b/cspell.json
@@ -13,6 +13,7 @@
     "contribs",
     "Daun",
     "Deskflow",
+    "Devs",
     "distro",
     "distros",
     "dmgbuild",
diff --git a/src/gui/src/dialogs/AboutDialog.cpp b/src/gui/src/dialogs/AboutDialog.cpp
index 9de836e8..c00722bb 100644
--- a/src/gui/src/dialogs/AboutDialog.cpp
+++ b/src/gui/src/dialogs/AboutDialog.cpp
@@ -41,28 +41,24 @@ AboutDialog::AboutDialog(QWidget *parent) : QDialog(parent)
   connect(btnCopyVersion, &QPushButton::clicked, this, [_version] { qApp->clipboard()->setText(_version); });
 
   auto versionLayout = new QHBoxLayout();
-  versionLayout->addSpacerItem(new QSpacerItem(50, 0, QSizePolicy::Fixed, QSizePolicy::Fixed));
   versionLayout->addWidget(new QLabel(tr("Version:")));
   versionLayout->addWidget(new QLabel(QString::fromStdString(deskflow::version())));
   versionLayout->addWidget(btnCopyVersion);
   versionLayout->addSpacerItem(new QSpacerItem(0, 0, QSizePolicy::Expanding, QSizePolicy::Fixed));
 
   auto lblLogo = new QLabel(this);
-  auto icon = QPixmap(deskflow::gui::isDarkMode() ? m_dark_logo : m_light_logo);
-  lblLogo->setPixmap(icon.scaled(250, 500, Qt::KeepAspectRatio, Qt::SmoothTransformation));
+  lblLogo->setPixmap(QPixmap(deskflow::gui::isDarkMode() ? m_darkLogo : m_lightLogo));
 
-  auto lblCopyright = new QLabel(tr("Copyright (C) 2024 Deskflow Developers\n"
-                                    "Copyright (C) 2012 - 2024 Symless Ltd.\n"
-                                    "Copyright (C) 2009 - 2012 Nick Bolton."));
+  auto lblCopyright = new QLabel(QString::fromStdString(deskflow::copyright()));
 
   auto boldFont = font();
   boldFont.setBold(true);
 
-  auto label_importantDevs = new QLabel(tr("Important developers"));
-  label_importantDevs->setFont(boldFont);
+  auto lblImportantDevsTitle = new QLabel(tr("Important developers"));
+  lblImportantDevsTitle->setFont(boldFont);
 
-  auto lbl_importantDevs = new QLabel(QStringLiteral("%1\n").arg(m_awesomeDevs.join(", ")));
-  lbl_importantDevs->setWordWrap(true);
+  auto lblImportantDevsBody = new QLabel(QStringLiteral("%1\n").arg(m_awesomeDevs.join(", ")));
+  lblImportantDevsBody->setWordWrap(true);
 
   auto btnOk = new QPushButton(tr("Ok"), this);
   btnOk->setDefault(true);
@@ -71,10 +67,10 @@ AboutDialog::AboutDialog(QWidget *parent) : QDialog(parent)
   auto mainLayout = new QVBoxLayout();
   mainLayout->addWidget(lblLogo);
   mainLayout->addLayout(versionLayout);
-  mainLayout->addWidget(new QLabel(tr("Keyboard and mouse sharing applicaion"), this));
+  mainLayout->addWidget(new QLabel(tr("Keyboard and mouse sharing application"), this));
   mainLayout->addWidget(lblCopyright);
-  mainLayout->addWidget(label_importantDevs);
-  mainLayout->addWidget(lbl_importantDevs);
+  mainLayout->addWidget(lblImportantDevsTitle);
+  mainLayout->addWidget(lblImportantDevsBody);
   mainLayout->addWidget(btnOk);
 
   setLayout(mainLayout);
diff --git a/src/gui/src/dialogs/AboutDialog.h b/src/gui/src/dialogs/AboutDialog.h
index 187d3800..fff42ec0 100644
--- a/src/gui/src/dialogs/AboutDialog.h
+++ b/src/gui/src/dialogs/AboutDialog.h
@@ -29,20 +29,25 @@ public:
   ~AboutDialog() = default;
 
 private:
-  inline static const auto m_light_logo = QStringLiteral(":/image/logo-light.png");
-  inline static const auto m_dark_logo = QStringLiteral(":/image/logo-dark.png");
+  inline static const auto m_lightLogo = QStringLiteral(":/image/logo-light.png");
+  inline static const auto m_darkLogo = QStringLiteral(":/image/logo-dark.png");
   inline static const auto m_awesomeDevs = QStringList{
       // Chris is the ultimate creator, and the one who started it all in 2001.
       QStringLiteral("Chris Schoeneman"),
+
       // Richard and Adam developed CosmoDeskflow, the 90's predecessor project.
       QStringLiteral("Richard Lee"),
       QStringLiteral("Adam Feder"),
+
       // Nick continued the legacy in 2009 started by Chris.
       QStringLiteral("Nick Bolton"),
+
       // Volker wrote the first version of the GUI (QSynergy) in 2008.
       QStringLiteral("Volker Lanz"),
+
       // Re-ignited the project in 2008 and rebuilt the community.
       QStringLiteral("Sorin Sbârnea"),
+
       // Contributors of bug fixes in the early days.
       QStringLiteral("Ryan Breen"),
       QStringLiteral("Guido Poschta"),
@@ -51,9 +56,11 @@ private:
       QStringLiteral("Brent Priddy"),
       QStringLiteral("Jason Axelson"),
       QStringLiteral("Jake Petroules"),
+
       // Implemented Wayland support (libei and libportal).
       QStringLiteral("Peter Hutterer"),
       QStringLiteral("Olivier Fourdan"),
+
       // Symless employees (in order of joining).
       QStringLiteral("Kyle Bloom"),
       QStringLiteral("Daun Chung"),
@@ -62,5 +69,8 @@ private:
       QStringLiteral("Olena Kutytska"),
       QStringLiteral("Owen Phillips"),
       QStringLiteral("Daniel Evenson"),
+
+      // Prominent Deskflow development community members
+      QStringLiteral("Chris Rizzitello"),
   };
 };
```

**Comment by sithlord48** [comment]
> ```diff
> -  versionLayout->addSpacerItem(new QSpacerItem(50, 0, QSizePolicy::Fixed, QSizePolicy::Fixed));
> ```

So we want that version label tucked all the way to the left ? Fine with me 

> ```diff
> +    "Copyright (C) 2024-%s Deskflow Developers\n"
> ```

This is not how copyright works you only change them for the years you made changes.

**Comment by sithlord48** [comment]
> ```diff
> -  mainLayout->addWidget(label_importantDevs);
> ```

>label QLabels that are a static label

Style guide

**Comment by nbolton** [comment]
> > ```diff
> > -  lblLogo->setPixmap(icon.scaled(250, 500, Qt::KeepAspectRatio, Qt::SmoothTransformation));
> > ```
> 
> This logo is a tad small so i bumped it up we can keep it at 100% size

Hmm, that makes it fuzzy. I think it's better to rezize the original image rather than forcing it to be larger. As it's not an SVG, it doesn't scale very well.

**Comment by nbolton** [comment]
> Style guide

Statics are `s_`

**Comment by sithlord48** [comment]
> Ok, now it's working but maybe logo is a tad fuzzy?
> 
> Hmm, also the copy button isn't showing for me. I'm on GNOME, not sure if relevant.
> 
> image

Looks like ill have to add a fallback to the built in qt one

**Review by nbolton — CHANGES_REQUESTED** [review]
Would you like me to amend your commit with these changes? 
- The `copyright` function can now be removed.
- Changed include ordering for safety (to avoid accidental include resolution)
- Changed name for copy button filename to `camelCase`
- Matched the statics to our code style (use `s_` instead of `m_`)
- Added some more context to the `isDarkMode` function docs

```diff
diff --git a/cspell.json b/cspell.json
index 166551ce..3e6be29f 100644
--- a/cspell.json
+++ b/cspell.json
@@ -13,6 +13,7 @@
     "contribs",
     "Daun",
     "Deskflow",
+    "Devs",
     "distro",
     "distros",
     "dmgbuild",
diff --git a/src/gui/src/dialogs/AboutDialog.cpp b/src/gui/src/dialogs/AboutDialog.cpp
index 90655691..422ed865 100644
--- a/src/gui/src/dialogs/AboutDialog.cpp
+++ b/src/gui/src/dialogs/AboutDialog.cpp
@@ -19,31 +19,30 @@
 
 #include "AboutDialog.h"
 
-#include <QClipboard>
-#include <QHBoxLayout>
-#include <QLabel>
-#include <QPushButton>
-#include <QStyle>
-
 #include "common/copyright.h"
 #include "common/version.h"
 #include "gui/style_utils.h"
 
+#include <QClipboard>
 #include <QDateTime>
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QPushButton>
+#include <QStyle>
 
 AboutDialog::AboutDialog(QWidget *parent) : QDialog(parent)
 {
   setWindowTitle(tr("About Deskflow"));
 
-  QString _version = QString::fromStdString(deskflow::version());
+  QString version = QString::fromStdString(deskflow::version());
 
   auto copyIcon = QIcon::fromTheme(
-      QIcon::ThemeIcon::EditCopy, deskflow::gui::isDarkMode() ? QIcon(m_darkcopy) : QIcon(m_lightcopy)
+      QIcon::ThemeIcon::EditCopy, deskflow::gui::isDarkMode() ? QIcon(s_darkCopy) : QIcon(s_lightCopy)
   );
 
   auto btnCopyVersion = new QPushButton(copyIcon, QString(), this);
   btnCopyVersion->setFlat(true);
-  connect(btnCopyVersion, &QPushButton::clicked, this, [_version] { qApp->clipboard()->setText(_version); });
+  connect(btnCopyVersion, &QPushButton::clicked, this, [version] { qApp->clipboard()->setText(version); });
 
   auto versionLayout = new QHBoxLayout();
   versionLayout->addWidget(new QLabel(tr("Version:")));
@@ -52,9 +51,9 @@ AboutDialog::AboutDialog(QWidget *parent) : QDialog(parent)
   versionLayout->addSpacerItem(new QSpacerItem(0, 0, QSizePolicy::Expanding, QSizePolicy::Fixed));
 
   auto lblLogo = new QLabel(this);
-  lblLogo->setPixmap(deskflow::gui::isDarkMode() ? m_darkLogo : m_lightLogo);
+  lblLogo->setPixmap(deskflow::gui::isDarkMode() ? s_darkLogo : s_lightLogo);
 
-  auto lblCopyright = new QLabel(QString::fromStdString(deskflow::copyright()));
+  auto lblCopyright = new QLabel(QString::fromStdString(deskflow::kCopyright));
 
   auto boldFont = font();
   boldFont.setBold(true);
@@ -62,7 +61,7 @@ AboutDialog::AboutDialog(QWidget *parent) : QDialog(parent)
   auto lblDevsTitle = new QLabel(tr("Important developers"));
   lblDevsTitle->setFont(boldFont);
 
-  auto lblDevsBody = new QLabel(QStringLiteral("%1\n").arg(m_awesomeDevs.join(", ")));
+  auto lblDevsBody = new QLabel(QStringLiteral("%1\n").arg(s_awesomeDevs.join(", ")));
   lblDevsBody->setWordWrap(true);
 
   auto btnOk = new QPushButton(tr("Ok"), this);
diff --git a/src/gui/src/dialogs/AboutDialog.h b/src/gui/src/dialogs/AboutDialog.h
index 63126323..0fc06733 100644
--- a/src/gui/src/dialogs/AboutDialog.h
+++ b/src/gui/src/dialogs/AboutDialog.h
@@ -29,22 +29,27 @@ public:
   ~AboutDialog() = default;
 
 private:
-  inline static const auto m_lightcopy = QStringLiteral(":/icons/64x64/copy-light.png");
-  inline static const auto m_darkcopy = QStringLiteral(":/icons/64x64/copy-dark.png");
-  inline static const auto m_lightLogo = QStringLiteral(":/image/logo-light.png");
-  inline static const auto m_darkLogo = QStringLiteral(":/image/logo-dark.png");
-  inline static const auto m_awesomeDevs = QStringList{
+  inline static const auto s_lightCopy = QStringLiteral(":/icons/64x64/copy-light.png");
+  inline static const auto s_darkCopy = QStringLiteral(":/icons/64x64/copy-dark.png");
+  inline static const auto s_lightLogo = QStringLiteral(":/image/logo-light.png");
+  inline static const auto s_darkLogo = QStringLiteral(":/image/logo-dark.png");
+  inline static const auto s_awesomeDevs = QStringList{
       // Chris is the ultimate creator, and the one who started it all in 2001.
       QStringLiteral("Chris Schoeneman"),
+
       // Richard and Adam developed CosmoDeskflow, the 90's predecessor project.
       QStringLiteral("Richard Lee"),
       QStringLiteral("Adam Feder"),
+
       // Nick continued the legacy in 2009 started by Chris.
       QStringLiteral("Nick Bolton"),
+
       // Volker wrote the first version of the GUI (QSynergy) in 2008.
       QStringLiteral("Volker Lanz"),
+
       // Re-ignited the project in 2008 and rebuilt the community.
       QStringLiteral("Sorin Sbârnea"),
+
       // Contributors of bug fixes in the early days.
       QStringLiteral("Ryan Breen"),
       QStringLiteral("Guido Poschta"),
@@ -53,9 +58,11 @@ private:
       QStringLiteral("Brent Priddy"),
       QStringLiteral("Jason Axelson"),
       QStringLiteral("Jake Petroules"),
+
       // Implemented Wayland support (libei and libportal).
       QStringLiteral("Peter Hutterer"),
       QStringLiteral("Olivier Fourdan"),
+
       // Symless employees (in order of joining).
       QStringLiteral("Kyle Bloom"),
       QStringLiteral("Daun Chung"),
@@ -64,6 +71,7 @@ private:
       QStringLiteral("Olena Kutytska"),
       QStringLiteral("Owen Phillips"),
       QStringLiteral("Daniel Evenson"),
+
       // Prominent Deskflow development community members
       QStringLiteral("Chris Rizzitello"),
   };
diff --git a/src/lib/common/copyright.h b/src/lib/common/copyright.h
index 183bffc0..15cf10b3 100644
--- a/src/lib/common/copyright.h
+++ b/src/lib/common/copyright.h
@@ -17,19 +17,12 @@
 
 #pragma once
 
-#include <string>
-
 namespace deskflow {
 
-const auto kCopyrightFormat = //
+const auto kCopyright = //
     "Copyright (C) 2024 Deskflow Developers\n"
     "Copyright (C) 2012-2024 Symless Ltd.\n"
     "Copyright (C) 2009-2012 Nick Bolton\n"
     "Copyright (C) 2002-2009 Chris Schoeneman";
 
-inline std::string copyright()
-{
-  return kCopyrightFormat;
-}
-
 } // namespace deskflow
diff --git a/src/lib/deskflow/App.cpp b/src/lib/deskflow/App.cpp
index 0a37aeb0..025ba510 100644
--- a/src/lib/deskflow/App.cpp
+++ b/src/lib/deskflow/App.cpp
@@ -94,13 +94,12 @@ App::~App()
 void App::version()
 {
   const auto version = deskflow::version();
-  const auto copyright = deskflow::copyright();
 
   const auto kBufferLength = 1024;
   std::vector<char> buffer(kBufferLength);
   std::snprintf(                                                   // NOSONAR
       buffer.data(), kBufferLength, "%s v%s, protocol v%d.%d\n%s", //
-      argsBase().m_pname, version.c_str(), kProtocolMajorVersion, kProtocolMinorVersion, copyright.c_str()
+      argsBase().m_pname, version.c_str(), kProtocolMajorVersion, kProtocolMinorVersion, deskflow::kCopyright
   );
 
   std::cout << std::string(buffer.data()) << std::endl;
diff --git a/src/lib/gui/style_utils.h b/src/lib/gui/style_utils.h
index 3fa5e9b0..bf82a3ef 100644
--- a/src/lib/gui/style_utils.h
+++ b/src/lib/gui/style_utils.h
@@ -24,9 +24,9 @@
 namespace deskflow::gui {
 
 /**
- * @brief Detects dark mode in a universal manor
- * the newer versions in qt 6.5+ are not always correct as it
- * needs platform support for it and some just return UnknownScheme
+ * @brief Detects dark mode in a universal manner (all Qt versions).
+ * Until better platform support is added, it's more reliable to use the old way (compare text and window lightness),
+ * because the newer versions in Qt 6.5+ are not always correct and some return `UnknownScheme`.
  * https://www.qt.io/blog/dark-mode-on-windows-11-with-qt-6.5
  */
 inline bool isDarkMode()
```

**Comment by sithlord48** [comment]
> Changed include ordering for safety (to avoid accidental include resolution)

What are you trying to avoid ? 

> Add spelling entry for "Devs"

Why is this included i do not see where we have added `Devs`

**Comment by sithlord48** [comment]
> > Please do not make reviews in the form of a large diffs. It makes the review much harder to follow.
> 
> Yes, I was wondering if it was a good idea or not. That answers that question!
> 
> I can find the suggestions feature a bit fiddly when there are many suggestions. Perhaps next time, opening a new PR with the original PR as a base could be a good approach. I might try that.

I would personally rather you comment on the offending line and put something like i would change this to 
```
whatever line how you like it 
```
then we can easily talk about each change and resolve as we go .

**Comment by nbolton** [comment]
> > Changed include ordering for safety (to avoid accidental include resolution)
> 
> What are you trying to avoid ?

Ordering includes in this way has advantages:
- Catch missing includes
- Avoid accidental reliance on system headers
- Promotes clean, self-contained headers

> > Add spelling entry for "Devs"
> 
> Why is this included i do not see where we have added `Devs`

`m_awesomeDevs` -- cspell works on camel case.

**Review by nbolton — CHANGES_REQUESTED** [review]
Changes requested in #7855 as doing this with the suggest feature would be fiddly.

**Review by nbolton — APPROVED** [review]

File: src/gui/src/dialogs/AboutDialog.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      QStringLiteral("Kyle Bloom"),
+      QStringLiteral("Daun Chung"),
+      QStringLiteral("Serhii Hadzhylov"),
+      QStringLiteral("Oleksandr Lysytsia"),
+      QStringLiteral("Olena Kutytska"),
+      QStringLiteral("Owen Phillips"),
+      QStringLiteral("Daniel Evenson"),
+      // Prominent Deskflow development community members
+      QStringLiteral("Chris Rizzitello"),
+  };
```
[the remark this answers is not in the document]
Comment (sithlord48) [reply src/gui/src/dialogs/AboutDialog.h:69]: Honestly i think that looks worse and makes it harder to follow we are setting a list.

[the remark this answers is not in the document]
Comment (nbolton) [reply src/gui/src/dialogs/AboutDialog.h]: @shymega Which way do you find more readable? With or without spaces. I'm happy to go with the majority.

**Review by nbolton — APPROVED** [review]

**Pull request merged**